docs/license-contributing/first-contributors-guide.md

```diff
@@ -0,0 +1,128 @@
+     ```
+
+
+## 2. **Set Up Your Local Development Environment**
+
+PSRule for Azure requires the following dependencies:
+
+- **PowerShell** (for running the tests and contributing to the codebase).
+- **PSRule.Rules.Azure** module (for testing Azure resources).
+- **Az PowerShell module** (for interacting with Azure resources).
+
+### Install Dependencies:
+
+1. Install the required modules from PowerShell Gallery:
+
+   ```
+   Install-Module -Name 'PSRule.Rules.Azure' -Scope CurrentUser
+   Install-Module -Name 'Az' -Scope CurrentUser
+   ```
+2. Install PSRule from GitHub if needed:
+ 
+    ```
+    Install-Module -Name 'PSRule' -Scope CurrentUser
+    ```
+
+
+## 3. **Understand the Project Structure**
+
+Before you start coding, it's helpful to understand the organization of the repository.
+
+
+## 4. **Contributing to Code**
+
+Before writing a fix or feature enhancement, ensure that an issue is logged. Be prepared to discuss your feature and take feedback. Additionally, ensure that your changes include unit tests and any necessary updates to the documentation.
+
+### Steps for Contributing Code:
+
+1. Fork the PSRule.Rules.Azure repo: Follow the steps above to fork and clone the repository.
+
+2. Create a new branch from `main` in your fork for your changes.
+
+3. Add commits in your branch, making sure they are logically grouped and well-described.
+
+4. If you have updated module code or rules, also update the `CHANGELOG.md`.
+    - Note: You don't need to update the `CHANGELOG.md` for changes to unit tests or documentation.
+
+5. Build your changes locally before pushing them. This ensures that they work as expected.
+
+6. Create a Pull Request (PR):
+    - Once you are ready for your changes to be reviewed, create a PR to merge changes into the PSRule `main` branch.
+    - If your changes are not ready for review, create a draft pull request instead.
+
+7. The Continuous Integration (CI) process will automatically build your changes. Ensure that your changes build successfully to be merged.
+    - If there are build errors, push new commits to your branch to fix them.
+
+8. Avoid using forced pushes or squashing changes while your PR is under review, as this makes it harder to review your changes.
+
+
+## 5. **Building and Testing Your Changes**
+
+Before opening a pull request, it’s important to build and test your changes locally. PSRule for Azure uses Continuous Integration (CI) pipelines to test changes across MacOS, Linux, and Windows configurations.
+
+1. Build and Test Locally:
+    - Ensure that you can build your changes locally. Follow the instructions in the [Building from Source](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/install.md#building-from-source) guide to set up your local environment for testing.
+
+2. Run All Tests:
+    - Before creating your PR, run the tests to make sure your changes are working as expected:
+      
+       ```
+       Invoke-Pester -Script .\tests
+       ```
+
```

Comment:
I think we are provide an example of a unit test for a rule here. Example is: https://github.com/Azure/psrule.rules.azure/blob/main/tests/PSRule.Rules.Azure.Tests/Azure.ACR.Tests.ps1#L40-L56